### ISSUE TITLE
DataChart Accessibility Audit

### ISSUE THREAD

**Issue by ShimiSun** [grommet/grommet#5324]
### Steps to Reproduce

1. 2. 3.

### Your Environment

- Grommet version:
- Browser Name and version:
- Operating System and version (desktop or mobile):

**Comment by amandadupell** [comment]
### Expected Behavior

The chart should relay some information about the data in the chart to users with screen readers.

### Actual Behavior

No data from the chart is relayed to the user effectively. User can tab to content but either no data is read aloud or the information is confusing and out of place.

### Video exhibiting the issue

https://user-images.githubusercontent.com/38971752/122128317-831e7580-ce02-11eb-9c76-a1799138dea6.mov

### Steps to Reproduce

1. Use VoiceOver for Mac Accessibility (or any other screen reader)
2. Open up the simple story DataChart in a new window from [Grommet Storybook](https://storybook.grommet.io/?path=/story/visualizations-datachart-simple--simple)
3. Use tab and key controls to go through chart and hear the information the screen reader relays

### Your Environment



- Grommet version: 2.17.3
- Browser Name and version: Chrome
- Operating System and version (desktop or mobile): MackBook Pro

**Comment by jcfilben** [comment]
Posting some notes and ideas Matt and I had around DataChart accessibility:

- DataChart should be presented with context. There should be a descriptive title and a summary. The summary should help provide the user with more context of what information they should be getting from the chart but the user should ultimately be able to drawn their own conclusions from looking at/interacting with the chart.
- If colors are used to convey information it is better to use shades of the same color rather than solid colors. If we were viewing the chart in grayscale we should be able to determine the different regions.
- Color should not be used as the only visual means of conveying information (WCAG 1.4.1).
- We should provide the user with different ways of viewing data. Give the user the option to view the data in either a chart mode or a tabular mode. This is beneficial to both screenreader users as well as users who have different ways of thinking/learning.
- If the chart is an svg there should be a short description and a long description. The short description identifies the image and tells the user how to find the long description.
- There may be certain charts where it makes sense to make them keyboard navigable. Highcharts has good examples of keyboard navigation within charts.
- If a chart is keyboard navigable, upon entering the chart area the screenreader should notify the user as well as inform them of the keystrokes they need to use to navigate the chart and how to leave the chart area.
- If a chart has tooltips that convey additional information we need to do additional research on the best way to make these accessible.

Ideas:
- It would be nice to develop a chart specification of how we think charts should be displayed (title, summary, chart options, legend, tabular view, etc.) and what the critical accessibility elements are.
- Can we bring the community into this, see if people are willing to engage and help us develop this specification.
- Maybe we can start with trying to make one storybook example accessible, test with bill and see what we learn.
- As things currently stand DataChart can be made to be accessible, although there are enhancements we would like to make in grommet to improve the accessibility experience. While we make enhancements it would be good to get some guidance and examples out there on can be done with grommet DataChart today to make it accessible.

Links:
w3.org/WAI/tutorials/images/complex/
https://www.w3.org/WAI/tutorials/images/examples/2014-first-qtr/
https://www.highcharts.com/blog/tutorials/best-chart-accessibility-practices/
https://www.elsevier.com/connect/making-charts-accessible-for-people-with-visual-impairments

**Comment by halocline** [comment]
The screenreader experience for this [chart example](https://www.highcharts.com/demo/line-basic) does a nice job of establishing context and allowing the user to choose to investigate the next level of detail or move on to the next content region. Overall, it gets much closer to the desired experience.
